test-generator.js

```diff
@@ -10,30 +13,58 @@ const formatOpts = {
   bracesSameLine: true
 };
 
-export const generateUnitTest = (msg) => {
-  if (!msg.data) {
+const leftHandDiff = (val) => {
+
+  if (isModifiedObject(val)) {
+    return val.__old;
+  }
+
+  if (isModifiedArray(val)) {
+    return val.filter(v => v.startsWith('~') || v.startsWith('-')).map(v => v.substring(1));
+  }
+
+  if (is(Object, val)) {
+    const nonAddedKeys = filter(s => !s.endsWith('__added'), keys(val));
```

Comment:
With stuff like this, it reads kinda awkwardly to go half in and half out on point-free style, so I'd recommend either one of the following:

```javascript
keys(val).filter(s => !s.endsWith('__added'))
```
...or...
```javascript
pipe(keys, reject(endsWith('__added')))(val);
```